packages/light-apps/src/IdentityHeader/IdentityHeader.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { Observable, Subscription } from 'rxjs';
+
+import { InputAddress } from './IdentityHeader.styles';
+
+// TODO: Add Governance Once That's in
+const APP_OPTIONS = [
+  {
+    key: 'Identity',
+    text: 'Identity',
+    value: 'Identity'
+  },
+  {
+    key: 'Transfer',
+    text: 'Transfer',
+    value: 'Transfer'
+  }];
+
+interface Props extends RouteComponentProps { }
+
+type State = {
+  balance?: Balance,
+  chain?: any
+};
+
+export class IdentityHeader extends React.PureComponent<Props, State> {
+  static contextType = AppContext;
+
+  context!: React.ContextType<typeof AppContext>; // http://bit.ly/typescript-and-react-context
+
+  state: State = {
+
+  };
+
+  balanceSub?: Subscription;
+  chainSub?: Subscription;
+
+  componentDidMount () {
+    this.subscribeBalance();
+    this.subscribeChain();
+  }
+
+  componentDidUpdate (prevProps: Props) {
+    if (prevProps.location.pathname.split('/')[2]
+        !== this.props.location.pathname.split('/')[2]) {
+      this.closeAllSubscriptions();
+      this.subscribeBalance();
+      this.subscribeChain();
+    }
+  }
+
+  closeAllSubscriptions () {
+    if (this.balanceSub) {
+      this.balanceSub.unsubscribe();
+      this.balanceSub = undefined;
+    }
+
+    if (this.chainSub) {
+      this.chainSub.unsubscribe();
+      this.chainSub = undefined;
+    }
+  }
+
+  getAddress = () => {
+    return this.props.location.pathname.split('/')[2];
+  }
+
+  getCurrentLocation = () => {
+    const { location } = this.props;
+
+    const currentLocation = location.pathname.split('/')[1].toLowerCase();
+
+    return currentLocation;
+  }
+
+  getName = () => {
+    const { keyring } = this.context;
+    const address = this.getAddress();
+
+    return address && keyring.getAccount(address).getMeta().name;
+  }
+
+  handleChangeCurrentAccount = (account: string) => {
+    const { history } = this.props;
+    const currentLocation = this.getCurrentLocation();
+
+    history.push(`/${currentLocation}/${account}`);
+  }
+
+  handleToggleApp = (event: React.SyntheticEvent<HTMLElement, Event>, { value }: DropdownProps) => {
+    const { history } = this.props;
+    const address = this.getAddress();
+
+    history.push(`/${value}/${address}`);
+  }
+
+  subscribeBalance = () => {
+    const { api } = this.context;
+    const currentAccount = this.getAddress();
+
+    // Subscribe to sender's balance
+    // FIXME using any because freeBalance gives a Codec here, not a Balance
+    // Wait for @polkadot/api to have TS support for all query.*
+    this.balanceSub = (api.query.balances.freeBalance(currentAccount) as Observable<Balance>)
+      .subscribe((balance) => this.setState({ balance }));
+  }
+
+  subscribeChain = () => {
+    const { api } = this.context;
+
+    this.chainSub = (api.rpc.system.chain()).subscribe(chain => this.setState({ chain }));
```

Comment:
I thought this might be useful elsewhere, in my previous PR I actually added this in the context: https://github.com/paritytech/substrate-light-ui/blob/master/packages/ui-common/src/AppContext.tsx#L22

Reply:
ok i'll use this to get this PR merged quickly, and bring in the ui-reactive components in future PRs